### PR TITLE
Rest traffic policy status correctly

### PR DESCRIPTION
**Pull request by ryantking**
Starting traffic policy status:

```yaml
status:
  observedGeneration: 1
  state: ACCEPTED
  trafficTargets:
    details-bookinfo-mgmt-cluster.gloo-mesh.:
      state: ACCEPTED
    reviews-bookinfo-mgmt-cluster.gloo-mesh.:
      state: ACCEPTED
  workloads:
  - details-v1-bookinfo-mgmt-cluster-deployment.gloo-mesh.
  - istio-ingressgateway-istio-system-mgmt-cluster-deployment.gloo-mesh.
  - productpage-v1-bookinfo-mgmt-cluster-deployment.gloo-mesh.
  - ratings-v1-bookinfo-mgmt-cluster-deployment.gloo-mesh.
  - reviews-v1-bookinfo-mgmt-cluster-deployment.gloo-mesh.
  - reviews-v2-bookinfo-mgmt-cluster-deployment.gloo-mesh.
  ```

Status after details service is deleted (pre-fix):

```yaml
status:
  errors:
  - TrafficTarget details.bookinfo.mgmt-cluster not found
  observedGeneration: 1
  state: INVALID
  trafficTargets:
    details-bookinfo-mgmt-cluster.gloo-mesh.:
      state: ACCEPTED
    reviews-bookinfo-mgmt-cluster.gloo-mesh.:
      state: ACCEPTED
  workloads:
  - details-v1-bookinfo-mgmt-cluster-deployment.gloo-mesh.
  - istio-ingressgateway-istio-system-mgmt-cluster-deployment.gloo-mesh.
  - productpage-v1-bookinfo-mgmt-cluster-deployment.gloo-mesh.
  - ratings-v1-bookinfo-mgmt-cluster-deployment.gloo-mesh.
  - reviews-v1-bookinfo-mgmt-cluster-deployment.gloo-mesh.
  - reviews-v2-bookinfo-mgmt-cluster-deployment.gloo-mesh.
  - reviews-v3-bookinfo-mgmt-cluster-deployment.gloo-mesh.
```

Status after details service is deleted (post-fix):

```yaml
status:
  errors:
  - TrafficTarget details.bookinfo.mgmt-cluster not found
  observedGeneration: 1
  state: INVALID
```

The traffic policy YAML I tested with is in the comments of the associated issue.
BOT NOTES: 
resolves https://github.com/solo-io/gloo-mesh/issues/1232

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -139,7 +139,7 @@ endif
 
 .PHONY: gloo-mesh-image-load
 gloo-mesh-image-load: gloo-mesh-image
-    kind load docker-image --name mgmt-cluster $(GLOOMESH_IMAGE):$(VERSION)
+	kind load docker-image --name mgmt-cluster $(GLOOMESH_IMAGE):$(VERSION)
```
Comment (joekelley) [inline Makefile:142]: let's revert this whitespace change

Comment (ryantking) [reply Makefile:142]: We actually need it since before the WS was spaces instead of a tab so `make` wasn't running the kind command.

Comment (harveyxia) [reply Makefile:142]: It's been working for me as well as CI—could you check any local settings of yours that might affect interpretation of white spaces?

File: changelog/v0.12.7/traffic_policy_status.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+changelog:
+  - type: FIX
+    issueLink: https://github.com/solo-io/gloo-mesh/issues/1232
+    description: Reset trafifc policy status correctly so deleted traffic targets are removed.
```
Comment (harveyxia) [inline changelog/v0.12.7/traffic_policy_status.yaml:4]:
```suggestion
    description: Reset traffic policy status correctly so deleted traffic targets are removed.
```

File: pkg/mesh-networking/apply/applier.go [hunk 3]
```diff
@@ -74,7 +73,7 @@ func initializePolicyStatuses(input input.LocalSnapshot) {
 		trafficPolicy.Status = networkingv1alpha2.TrafficPolicyStatus{
 			State:              networkingv1alpha2.ApprovalState_ACCEPTED,
 			ObservedGeneration: trafficPolicy.Generation,
-			TrafficTargets:     trafficPolicy.Status.TrafficTargets,
+			TrafficTargets:     map[string]*networkingv1alpha2.ApprovalStatus{},
```
Comment (harveyxia) [inline pkg/mesh-networking/apply/applier.go:76]: nice :)

File: Makefile (hunk 1 above)
Comment (harveyxia) [reply Makefile:142]: discussed offline, verified that this is a bug

**Review by harveyxia — APPROVED** [review]

**Pull request merged**